nilearn/experimental/surface/_surface_image.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import annotations
+
+import abc
+import dataclasses
+import pathlib
+from typing import Dict
+
+import numpy as np
+
+from nilearn.experimental.surface import _io
+
+PolyData = Dict[str, np.ndarray]
+
+
+class Mesh(abc.ABC):
+    """A surface mesh, having vertex coordinates and faces(triangles)."""
+
+    n_vertices: int
+
+    # TODO those are properties for now for compatibility with plot_surf_img
+    # for the demo but should become functions as they can take some time to
+    # return
+    coordinates: np.ndarray
+    faces: np.ndarray
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__} with {self.n_vertices} vertices>"
+
+
+class FileMesh(Mesh):
+    """A surface mesh stored in a Gifti or Freesurfer file."""
+
+    n_vertices: int
+
+    file_path: pathlib.Path
```

Comment:
the `__init__` should take care of making it a Path so we can then rely on it being a Path and so we deal with that in the init only instead of everywhere